lib/Common/Memory/HeapInfo.cpp

```diff
@@ -1766,12 +1766,61 @@ BOOL SmallAllocationBlockAttributes::IsAlignedObjectSize(size_t sizeCat)
 {
     return HeapInfo::IsAlignedSmallObjectSize(sizeCat);
 }
+/* static */
+uint SmallAllocationBlockAttributes::GetUnusablePageCount(size_t sizeCat)
+{
+    UNREFERENCED_PARAMETER(sizeCat);
+    return 0;
+}
+/* static */
+void SmallAllocationBlockAttributes::ProtectUnusablePages(HeapBlock* heapBlock)
+{
+    UNREFERENCED_PARAMETER(heapBlock);
+}
+/* static */
+BOOL SmallAllocationBlockAttributes::RestoreUnusablePages(HeapBlock* heapBlock)
+{
+    UNREFERENCED_PARAMETER(heapBlock);
+    return TRUE;
+}
 
 /* static */
 BOOL MediumAllocationBlockAttributes::IsAlignedObjectSize(size_t sizeCat)
 {
     return HeapInfo::IsAlignedMediumObjectSize(sizeCat);
 }
+/* static */
+uint MediumAllocationBlockAttributes::GetUnusablePageCount(size_t sizeCat)
+{
+    return ((MediumAllocationBlockAttributes::PageCount*AutoSystemInfo::PageSize) % sizeCat) / AutoSystemInfo::PageSize;
+}
+/* static */
+void MediumAllocationBlockAttributes::ProtectUnusablePages(HeapBlock* heapBlock)
```

Comment:
I'm not sure why this is here and not in the HeapBlock- GetUnusablePageCount can be argued as getting an attribute of the block, this is definitely not an attribute. Also, nitpicky but please fix spacing of usage of *- not sure why its the only operator that you use with no spaces